### PR TITLE
add GitLab CI integration Job

### PR DESCRIPTION
**Pull request by SimeonEhrig**
fix: #2

### CODE REVIEW

**Comment by SimeonEhrig** [comment]
Because the GitLab CI mirror is not working yet, mirror manual to GitLab: https://gitlab.com/hzdr/qedjl-project/integrationtest.jl-test-mirrow/-/tree/addGitLabCIExample?ref_type=heads

File: .ci/generateGitLabCI.jl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+using IntegrationTests
+using Pkg
+using PkgDependency
+
+include(joinpath(dirname(@__FILE__), "prepareIntegrationTest.jl"))
+
+"""
+    print_job_yaml(job_name::AbstractString)
+
+Generate GitLab CI job yaml for a given package name and print it stdout.
```
Comment (AntonReinhard) [inline .ci/generateGitLabCI.jl:12]:
```suggestion
Generate a GitLab CI job yaml file for a given package name and print it to stdout.
```

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+stages:
+  - generator
+  - runTests
+
+generateIntegrationTests:
+  stage: generator
+  image: "julia:1.9"
```
Comment (AntonReinhard) [inline .gitlab-ci.yml:7]: Should probably use 1.10?
```suggestion
  image: "julia:1.10"
```

File: docs/src/pipeline_tutorials.md [hunk 3]
```diff
@@ -4,7 +4,52 @@ This section shows how to set up CI tests for different platforms with `Integrat
 
 ## GitLab CI
 
-Coming soon.
+GitLab CI offers the [parent-child-pipeline](https://docs.gitlab.com/ee/ci/pipelines/downstream_pipelines.html#parent-child-pipelines) function. This allows you to generate GitLab CI job yaml code in a GitLab CI job and start the generated jobs in the same pipeline.
+
+
+For our example, let's assume that we use a Julia script called `generateIntegrationTests.jl` that uses `IntegrationTests::depending_projects()` to generate the integration tests. This script generates valid GitLab CI yaml code and outputs it to stdout. The GitLab CI code in .gitlab-ci.yml could look like this:
```
Comment (AntonReinhard) [inline docs/src/pipeline_tutorials.md:10]:
```suggestion
For our example, let's assume that we use a Julia script called `generateIntegrationTests.jl` that uses `IntegrationTests::depending_projects()` to generate the integration tests. This script generates valid GitLab CI yaml code and outputs it to stdout. The GitLab CI code in `.gitlab-ci.yml` could look like this:
```

File: docs/src/pipeline_tutorials.md [hunk 4]
```diff
@@ -4,7 +4,52 @@ This section shows how to set up CI tests for different platforms with `Integrat
...
+
+```yaml
+stages:
+  - generator
+  - runTests
+
+generateIntegrationTests:
+  stage: generator
+  image: "julia:1.9"
```
Comment (AntonReinhard) [inline docs/src/pipeline_tutorials.md:19]:
```suggestion
  image: "julia:1.10"
```

File: docs/src/pipeline_tutorials.md [hunk 5]
```diff
@@ -35,7 +80,7 @@ jobs:
         - uses: actions/checkout@v3
         - uses: julia-actions/setup-julia@v1
             with:
-            version: 1.9
+            version: "1.9"
```
Comment (AntonReinhard) [inline docs/src/pipeline_tutorials.md:83]:
```suggestion
            version: "1.10"
```

**Review by AntonReinhard — APPROVED** [review]
Looks good I think, just some minor suggestions

File: .gitlab-ci.yml (hunk 2 above)
Comment (SimeonEhrig) [reply .gitlab-ci.yml:7]: Actual yes, but I don't test Julia 1.10 yet. I want to add Julia 1.10 support in an extra PR. And there, I will also increase the Julia versions for jobs like this.

Comment (AntonReinhard) [reply .gitlab-ci.yml:7]: :+1

**Pull request merged**